transaction/std/src/transaction_builder.rs

```diff
@@ -463,8 +464,8 @@ fn create_output_with_fog_hint<RNG: CryptoRng + RngCore>(
     rng: &mut RNG,
 ) -> Result<(TxOut, RistrettoPublic), TxBuilderError> {
     let private_key = RistrettoPrivate::from_random(rng);
-    let mut tx_out =
-        TxOut::new_with_memo(value, token_id, recipient, &private_key, fog_hint, memo_fn)?;
+    let amount = Amount { value, token_id };
```

Comment:
How about we make the `create_output*` functions also take an `Amount`?

Reply:
so i think the thing here was, the token id is a config parameter to the tx builder, and so it is implied. so that was why it was passing `value: u64` around. We could still make it take `Amount` and then double check the token ids I guess.